airgun/entities/cloud_inventory.py

```diff
@@ -11,26 +13,31 @@ class CloudInventoryEntity(BaseEntity):
 
     def read(self, entity_name=None, widget_names=None):
         view = self.navigate_to(self, 'All')
+        result = view.read(widget_names=widget_names)
         if entity_name:
-            view.searchbox.search(entity_name)
-            entity_item = next(view.inventory_list[entity_name])
-            return entity_item.read()
-        view.searchbox.clear()
-        return view.read(widget_names=widget_names)
+            view.inventory_list.toggle(entity_name)
+            result.update(view.inventory_list.read())
+        return result
 
     def generate_report(self, entity_name):
         view = self.navigate_to(self, 'All')
-        view.searchbox.search(entity_name)
-        entity_item = next(view.inventory_list[entity_name])
-        entity_item.browser.click(entity_item.generating.restart, ignore_ajax=True)
-        wait_for(lambda: entity_item.status == 'idle', timeout=180, delay=1, logger=view.logger)
+        view.inventory_list.toggle(entity_name)
+        view.browser.click(view.inventory_list.generating.restart, ignore_ajax=True)
+        # Fix
```

Comment:
Please provide some more depth here - will this comment help you when you read it 6 months from now?  Is this a fix, or does this need to be fixed?

What is this fixing? Why must we sleep here, even though we're using `wait_for`? What about the page, browser, or Satellite UI is causing the need to have a static sleep here? (5 seconds its ages to statically wait in an automation pipeline) What have you tried to work around it that didn't work?

The comment should answer these types of questions, so that the next person reading it can understand what's going on - and maybe on day fix it.

Sometimes providing all the context in-line gets too verbose, in which case its good to add more explanation to the commit. Someone reading this in the future has a reasonable chance of knowing to read the commit that `git blame` identifies for this line.

Reply:
You are right should have mentioned the issue before, will keep this in mind for future issues. here's what the problem is:
When we click on `restart` button then the process of generating and uploading a report gets started. There are two icons one for generating tab and other for uploading tab. They indicate users whether process of uploading or generating report is finished or in progress. So we were using those icon to wait till reports are generated and uploaded based on `entity_item.status` can result in `uploading`, `generating` and `idle` state. But the issue is that those icon don't change their state immediately after we start report generation process(clicking on `restart` button), which makes automation think that the process is finished(`idle`), which is not correct. That's why we have added sleep to wait for icons to change their state(`generating` or `uploading`). Also the process of generating/uploading reports takes some time either way, so I thought we can wait for few seconds.

Reply:
Thanks @jameerpathan111